### ISSUE TITLE
Surface custom fields in Woo App.

### ISSUE THREAD

**Issue by senadir** [woothemes/woocommerce#42139]
Work with the Woo App team to see how we can surface custom fields in the Woo App order view and possibly the create order view if needed.

For now, we might need design input on this as well, but it can go last.

The idea here is to make sure the Woo App team has the tools and knowledge they need to surface those fields and/or save them.

**Comment by kkoteen** [comment]
@senadir I'm cleaning up some old issues. I believe this work is complete. Attaching screenshots here. Let me know if there's additional work that needs to be done. Thanks! 

<img width="590" height="1278" alt="Image" src="https://github.com/user-attachments/assets/425cd5c8-5d80-4615-82b0-22b332774474" />
<img width="590" height="1278" alt="Image" src="https://github.com/user-attachments/assets/cd69d151-5d9f-4468-bd27-7321b5d4ab44" />
<img width="1179" height="2556" alt="Image" src="https://github.com/user-attachments/assets/9424ca7b-6714-4bc7-9766-0a57db9a7d6a" />

**Comment by senadir** [comment]
Nope, that's meta fields, but we're looking to expose additional fields (I updated the issue title to reflect the actual naming we landed on). Additional fields are text ones collected during checkout, and they automatically surface on the order screen, in My Account, and in emails
https://developer.woocommerce.com/docs/block-development/extensible-blocks/cart-and-checkout-blocks/additional-checkout-fields/

**Comment by kkoteen** [comment]
Thanks @senadir! Moving this into Linear: https://linear.app/a8c/issue/WOOMOB-3331/surface-additional-checkout-fields-in-woo-app
